web/api/v1/api_test.go

```diff
@@ -2171,6 +2171,69 @@ func TestRespondError(t *testing.T) {
 	}
 }
 
+func TestParseTimeParam(t *testing.T) {
+	type resultType struct {
+		asTime  time.Time
+		asError func() error
+	}
+	ts, err := parseTime("1582468023986")
+
+	testutil.Ok(t, err)
+
+	var tests = []struct {
+		paramName    string
+		paramValue   string
+		defaultValue time.Time
+		result       resultType
+	}{
+		{ // When data is valid.
+			paramName:    "start",
+			paramValue:   "1582468023986",
+			defaultValue: minTime,
+			result: resultType{
+				asTime:  ts,
+				asError: nil,
+			},
+		},
+		{ // When data is empty string.
+			paramName:    "end",
+			paramValue:   "",
+			defaultValue: maxTime,
+			result: resultType{
+				asTime:  maxTime,
+				asError: nil,
+			},
+		},
+		{ // When data is not valid.
+			paramName:    "foo",
+			paramValue:   "baz",
+			defaultValue: maxTime,
+			result: resultType{
+				asTime: time.Time{},
+				asError: func() error {
+					_, err := parseTime("baz")
+					return errors.Wrapf(err, "Invalid time value for '%s'", "foo")
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		req, err := http.NewRequest("GET", "localhost:42/foo?"+test.paramName+"="+test.paramValue, nil)
+
+		testutil.Ok(t, err)
+
+		result := test.result
+		asTime, err := parseTimeParam(req, test.paramName, test.defaultValue)
+
+		if err != nil {
+			testutil.ErrorEqual(t, result.asError(), err)
+		} else {
+			testutil.Assert(t, asTime.Equal(result.asTime), "time as return value: %s not parsed correctly. Expected %s. Actual %s", test.paramValue, result.asTime, asTime)
+		}
+
```

Comment:
unneeded blank line :)

Reply:
I ran the ```go fmt``` command but looks like it's not working for blank lines